### PR TITLE
Use unbuffered IO for stdout in NIOPerformanceTester

### PR DESCRIPTION
**Pull request by simonjbeaumont**
### Motivation:

While updating `NIOPerformanceTester` in #2063 a test was crashing but it was hard to see which test was running at the time of the crash due to output buffering which meant we had to tediously binary search for the test that was causing the CI crash.

### Modifications:

Switch to using unbuffered IO for standard output within `NIOPerformanceTester`.

### Result:

Output from `NIOPerformanceTester` will be immediately flushed so that, in the event of a crash in CI, it will be clear which test was running.

### CODE REVIEW

**Comment by simonjbeaumont** [comment]
//cc @tomerd @Lukasa

**Comment by simonjbeaumont** [comment]
@swift-nio-bot test perf please

**Comment by simonjbeaumont** [comment]
Note, I didn't run the perf test because I think this change should improve performance, just to check it doesn't break the harness itself.

**Comment by simonjbeaumont** [comment]
Let me test that this _actually_ helps by causing the same crash as we had in #2063. Standby...

**Comment by simonjbeaumont** [comment]
@swift-nio-bot test perf please

**Comment by simonjbeaumont** [comment]
Here's the CI run with the same crash as we had in #2063: https://ci.swiftserver.group/job/swift-nio2-performance-prb/131/

From the logs, we can now see exactly which test it was that failed, the `schedule_1m_tasks`:

```
11:43:30 measuring: schedule_1m_tasks: Build step 'Execute shell' marked build as failure
```

I'll now revert that commit so it can be squashed away in the merge.

**Comment by tomerd** [comment]
seems reasonable to me 👍

**Comment by simonjbeaumont** [comment]
@swift-nio-bot test this please

File: Sources/NIOPerformanceTester/main.swift [hunk 1]
```diff
@@ -19,6 +19,9 @@ import NIOFoundationCompat
 import Dispatch
 import NIOWebSocket
 
+// Use unbuffered stdout to help detect exactly which test was running in the event of a crash.
+setbuf(stdout, nil)
+
```
Comment (adam-fowler) [inline Sources/NIOPerformanceTester/main.swift:24]: Not sure this is relevant in this case, but macOS outputs test results to stderr

Comment (simonjbeaumont) [reply Sources/NIOPerformanceTester/main.swift:24]: Hmm. This isn’t an XCTest thing; this is just a Swift binary with a main.swift using `print`. Are you sure that goes to stderr on macOS?

Comment (adam-fowler) [reply Sources/NIOPerformanceTester/main.swift:24]: Ah sorry. Should have realised that.

Comment (simonjbeaumont) [reply Sources/NIOPerformanceTester/main.swift:24]: That’s good to know for future, thanks!

**Review by Lukasa — APPROVED** [review]

**Pull request merged**